http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/sender/TargetHandler.java

```diff
@@ -83,8 +83,6 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                     HttpContent httpContent = (HttpContent) msg;
                     targetRespMsg.addHttpContent(httpContent);
                     if (msg instanceof LastHttpContent) {
-                        targetRespMsg.setEndOfMsgAdded(true);
-                        targetRespMsg = null;
                         if (HTTPTransportContextHolder.getInstance().getHandlerExecutor() != null) {
```

Comment:
As a practice, get the value of "HTTPTransportContextHolder.getInstance().getHandlerExecutor()" to a variable, and use the variable after that, rather than doing the same lookup twice. It may or may not be an expensive operation, but still, better for readability, and if in the critical path, can be better performance wise, if the lookup happens to be expensive.